hieroplatform/src/test/java/org/hiero/sketch/NextKSketchTest.java

```diff
@@ -0,0 +1,112 @@
+
+public class NextKSketchTest {
+
+    @Test
+    public void testTopK1() {
+        final int numCols = 2;
+        final int maxSize = 50;
+        final int rightSize = 1000;
+        final int leftSize = 1000;
+        final Table leftTable = getRepIntTable(leftSize, numCols);
+        //System.out.println(leftTable.toLongString(50));
+        RecordOrder cso = new RecordOrder();
+        for (String colName : leftTable.getSchema().getColumnNames()) {
+            cso.append(new ColumnSortOrientation(leftTable.getSchema().getDescription(colName), true));
+        }
+        final RowSnapshot topRow = new RowSnapshot(leftTable, 10);
+        //System.out.printf("Top Row %s. %n", topRow.toString());
+        final NextKSketch nk = new NextKSketch(cso, topRow, maxSize);
+        final NextKList leftK = nk.getNextKList(leftTable);
+        IndexComparator leftComp = cso.getComparator(leftK.table);
+        for (int i = 0; i < (leftK.table.getNumOfRows() - 1); i++)
+            assertTrue(leftComp.compare(i, i + 1) <= 0);
+        //System.out.println(leftK.toLongString(maxSize));
+
+        final RowSnapshot topRow2 = new RowSnapshot(leftTable, 100);
+        //System.out.printf("Top Row %s. %n", topRow2.toString());
+        final NextKSketch nk2 = new NextKSketch(cso, topRow2, maxSize);
+        final NextKList leftK2 = nk2.getNextKList(leftTable);
+        IndexComparator leftComp2 = cso.getComparator(leftK2.table);
+        for (int i = 0; i < (leftK2.table.getNumOfRows() - 1); i++)
+            assertTrue(leftComp2.compare(i, i + 1) <= 0);
+        //System.out.println(leftK2.toLongString(maxSize));
+        final Table rightTable = getRepIntTable(rightSize, numCols);
+        final NextKList rightK = nk.getNextKList(rightTable);
+        IndexComparator rightComp = cso.getComparator(rightK.table);
+        for (int i = 0; i < (rightK.table.getNumOfRows() - 1); i++)
+            assertTrue(rightComp.compare(i, i + 1) <= 0);
+        //System.out.println(rightK.toLongString(maxSize));
+        final NextKList tK = nk.add(leftK, rightK);
+        IndexComparator tComp = cso.getComparator(tK.table);
+        for (int i = 0; i < (tK.table.getNumOfRows() - 1); i++)
+            assertTrue(tComp.compare(i, i + 1) <= 0);
+        //System.out.println(tK.toLongString(maxSize));
+    }
+
+
+    public void testTopK2() {
+        final int numCols = 2;
+        final int maxSize = 50;
+        final int rightSize = 1000;
+        final int leftSize = 1000;
+        final Table leftTable = getRepIntTable(leftSize, numCols);
+        final RowSnapshot topRow = new RowSnapshot(leftTable, 10);
+        //System.out.println(leftTable.toLongString(50));
+        RecordOrder cso = new RecordOrder();
+        final NextKSketch nk= new NextKSketch(cso, topRow, maxSize);
+        final NextKList leftK = nk.getNextKList(leftTable);
+        IndexComparator leftComp = cso.getComparator(leftK.table);
+        System.out.println(leftK.toLongString(maxSize));
```

Comment:
Would be nice to replace this with some kind of assertion. Same comment for next method.